### PR TITLE
Feature/change password

### PR DESCRIPTION
**Pull request by JanuszBartosz**
Change password functionality.

### CODE REVIEW

**Comment by adriankulinski** [comment]
Wzorcowy kod Bartek :)

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/PasswordValidator.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	public boolean supports(Class<?> clazz) {
+		
+		return clazz.equals(PasswordDTO.class);
+	}
+
+	//Metoda ustawiająca parametry validatora.
+	@Override
+	public void validate(Object target, Errors errors) {
+		PasswordDTO passwordDTO = (PasswordDTO) target;
+		validatePassword(errors, passwordDTO);
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/PasswordValidator.java]: Good!

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/PasswordValidator.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+		
+	}
+	
+	//Warunki validacji.
+	private void validatePassword(Errors errors, PasswordDTO passwordDTO) {
+		
+		if( !(passwordDTO.getOldPassword() != null && !passwordDTO.getOldPassword().isEmpty()) )
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/PasswordValidator.java]: Please use StringUtils for empty string check (or wright by your self) and get rid of repeated code.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/PasswordValidator.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+		if( !(passwordDTO.getNewPassword() != null && !passwordDTO.getNewPassword().isEmpty()) )
+			errors.reject("new.password.empty", "New password is empty!");
+		
+		if( !(passwordDTO.getNewPasswordRepeat() != null && !passwordDTO.getNewPasswordRepeat().isEmpty()) )
+			errors.reject("repeat.password.empty", "Repeat password is empty!");
+				
+		if( !(passwordDTO.getNewPasswordRepeat() != null && passwordDTO.getNewPassword().equals(passwordDTO.getNewPasswordRepeat())) )
+			errors.reject("passwords.not.equal", "Password and repeated password do not match!");
+					
+		String email = (String) session.getAttribute("email");
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/PasswordValidator.java]: Please retrieve user from Spring Security Context rather then from session attribute and remove Session class attribute

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/PasswordValidator.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+							
+		User user = userService.getUserByEmail(email).get();
+		String oldPasswordHash = user.getPassword();
+		
+		if(! new BCryptPasswordEncoder().matches(passwordDTO.getOldPassword(), oldPasswordHash))
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/PasswordValidator.java]: Please change encoder to bean (create in Security Config) and inject it here

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/SecurityConfig.java [hunk 5]
```diff
@@ -46,6 +50,8 @@ protected void configure(HttpSecurity http) throws Exception {
 							
 	}
 	
+	
+	//Uruchomienie silnika Spring Security.
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/SecurityConfig.java]: Please leave comments in English

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/LoginController.java [hunk 6]
```diff
@@ -9,23 +9,22 @@
 import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.bind.annotation.ResponseBody;
 
-import com.pgs.soft.service.impl.UserServiceImpl;
+import com.pgs.soft.service.UserService;
 
 @Controller
 @ResponseBody
 public class LoginController {
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/LoginController.java]: This controller is not needed (all login acctions is made by Spring Security)

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/RegisterController.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+		this.registerValidator = userDtoValidator;
+	}
+	
+	@InitBinder
+    public void initBinder(WebDataBinder binder) {
+        binder.addValidators(registerValidator);
+    }
+	
+	@RequestMapping(value = "/user", method=RequestMethod.POST)
+	public String register(@Valid @RequestBody UserDTO userDTO){
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/RegisterController.java:37]: Please move it to UserController

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/dto/PasswordDTO.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
+package com.pgs.soft.dto;
+
+public class PasswordDTO {
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/dto/PasswordDTO.java]: Please change name to ChangePasswordRequestDTO - name should said exactly what class is responsible for

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/CustomAuthenticationSuccessHandler.java [hunk 9]
```diff
@@ -24,6 +24,7 @@ public void onAuthenticationSuccess(HttpServletRequest httpServletRequest, HttpS
 
         HttpSession session = httpServletRequest.getSession();
         User authUser = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        session.setAttribute("id", authUser.getId());
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/CustomAuthenticationSuccessHandler.java]: What for are those lines needed?

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/UserServiceImpl.java [hunk 10]
```diff
@@ -19,8 +22,13 @@
 public class UserServiceImpl implements UserService, UserDetailsService{
 	
 	@Autowired
+	HttpSession session;
+	
+	@Autowired
 	UserRepository userRepository;
 	
+	private BCryptPasswordEncoder bCryptPasswordEncoder = new BCryptPasswordEncoder();
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/UserServiceImpl.java]: Please change it to bean

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/PasswordValidator.java [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+
+	@Autowired
+	private HttpSession session;
+	
+	@Autowired 
+	private UserService userService;
+	
+	@Override
+	public boolean supports(Class<?> clazz) {
+
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/PasswordValidator.java]: Please remove blank line :)

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/PasswordValidator.java (hunk 2 above)
Comment (JanuszBartosz) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/PasswordValidator.java]: Sure, I read about StringUtils, but it seemed to me a little too strong for just this operation.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/UserServiceImpl.java (hunk 10 above)
Comment (JanuszBartosz) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/UserServiceImpl.java]: Done.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/dto/PasswordDTO.java (hunk 8 above)
Comment (JanuszBartosz) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/dto/PasswordDTO.java]: Done.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/CustomAuthenticationSuccessHandler.java (hunk 9 above)
Comment (JanuszBartosz) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/CustomAuthenticationSuccessHandler.java]: Session deleted from SuccessHandler.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/PasswordValidator.java (hunk 11 above)
Comment (JanuszBartosz) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/PasswordValidator.java]: Done.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/RegisterController.java (hunk 7 above)
Comment (JanuszBartosz) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/RegisterController.java:37]: This one is for Adrian,

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/LoginController.java (hunk 6 above)
Comment (JanuszBartosz) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/controller/LoginController.java]: Controller deleted.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/PasswordValidator.java (hunk 3 above)
Comment (JanuszBartosz) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/PasswordValidator.java]: Done.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/PasswordValidator.java (hunk 4 above)
Comment (JanuszBartosz) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/PasswordValidator.java]: Done.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/SecurityConfig.java (hunk 5 above)
Comment (JanuszBartosz) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/SecurityConfig.java]: Done.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/PasswordValidator.java (hunk 2 above)
Comment (JanuszBartosz) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/PasswordValidator.java]: Done.

**Pull request merged**